web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/verge/VergeRestTjeneste.java

```diff
@@ -62,11 +68,47 @@ public VergeRestTjeneste(BehandlingsprosessTjeneste behandlingsprosessTjeneste,
         this.vergeTjeneste = vergeTjeneste;
     }
 
+    @GET
+    @Operation(description = "Henter verge/fullmektig på behandlingen", tags = "verge", responses = {@ApiResponse(responseCode = "200", description = "Verge/fullmektig funnet"), @ApiResponse(responseCode = "204", description = "Ingen verge/fullmektig")})
+    @BeskyttetRessurs(actionType = ActionType.READ, resourceType = ResourceType.FAGSAK)
+    public VergeDto hentVerge(@QueryParam(BehandlingsUuidParam.NAME) @Valid BehandlingsUuidParam queryParam) {
+
+        var behandling = behandlingsprosessTjeneste.hentBehandling(queryParam.getBehandlingUuid());
+        return vergeTjeneste.hentVerge(behandling);
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Operation(description = "Oppretter verge/fullmektig på behandlingen", tags = "verge", responses = {@ApiResponse(responseCode = "202", description = "Verge/fullmektig opprettes", headers = @Header(name = HttpHeaders.LOCATION))})
+    @BeskyttetRessurs(actionType = ActionType.UPDATE, resourceType = ResourceType.FAGSAK)
+    public Response opprettVerge(@Context HttpServletRequest request,
+                                 @QueryParam(BehandlingsUuidParam.NAME) @Valid BehandlingsUuidParam queryParam,
+                                 @Valid NyVergeDto body) throws URISyntaxException {
+
+        var behandling = behandlingsprosessTjeneste.hentBehandling(queryParam.getBehandlingUuid());
+        vergeTjeneste.opprettVerge(behandling, body);
+
+        return Redirect.tilBehandlingPollStatus(request, queryParam.getBehandlingUuid());
+    }
+
+    @DELETE
```

Comment:
Delete kan virke som inituitiv METHOD - men et par ting
- vi sletter ikke informasjon - vi setter den som ikke aktiv - slik at vi har historikken i databasen. Dvs endring
- Semantikk: Vi "oppdaterer" behandlingen men "sletter" verge (eller setter inaktiv) 
- Hva er ressursen her ? Behandlingen er førsteordens med referanse, verge er eksistensavhengig

Reply:
Enig i at det er en sematisk forskjell. Kan endres til PUT for eksempel.

> Hva er ressursen her ? Behandlingen er førsteordens med referanse, verge er eksistensavhengig

Resurssen er endepunktet. For å legge inn eksistensavhengigheten kan vi legge til dette i BehandlingDTOTjeneste:
```java        
if (vergeRepository.hentAggregat(behandling.getId()).isPresent()) {
        dto.leggTil(delete(VergeRestTjeneste.BASE_PATH, "verge-fjern", new BehandlingsUuidParam(behandling.getUuid())));
} else {
        dto.leggTil(post(VergeRestTjeneste.BASE_PATH, "verge-opprett", new BehandlingsUuidParam(behandling.getUuid()), new NyVergeDto()));
}
```
Tanker?

Reply:
Vi har 1 PUT (opprett behandling), ingen PATCH og ingen DELETE. 
Rest-verden klarer seg i stor grad med GET og POST - de andre metodene er i langt mindre bruk. 
Jeg vet om et par interne API som bruker pathparams + alle P-metodene + delete. Det var sikkert en lur tanke bak, men jeg er usikker på verdien det gir. PATCH er tross alt ikke med som standard i java.net.http.HttpRequest
Derfor - jeg ville nok gjort det enkelt med POST fram til teamet er med på en omforent endring i praksis.

Reply:
Det er et alternativ å standardisere på kun POST for å ta ned risk for Q-param-lekkasje av identer til logger. 
Vi har tross alt stort sett bare interne APIs

Reply:
Jeg synes ikke vi skal standarisere på POST uten queryparametere sånn helt uten videre. Det er en kjempefordel at vi i frontend får sendt ned ressurslenker med enten query(eller path) parametere, det gjør at vi veldig enket kan sende request uten at vi trenger å manuelt populere ting som behandlingsuuid eller saksnummer. Parametere som er sensitive vil vi i utganspunktet hindre at blir lagt til som query parameters gjennom kodegjennomgang og praksis, og eventuelt generelle RestTjeneste tester. Detter tror jeg vi skal diskutere mer en gang vi er på kontoret

Reply:
Vi har mange ting som bør gjøres i grenseland frontend/backend og jeg tenker vi bør lage en bredere aktivitet og få på bordet problem, tilta, målbilde og gjennomføring. Norske bokstaver, JsonProperty, Polling-mekanismen, PathParams, HTTP-method, lenker, kodeverk, ubrukte felt, etc. 
Vi har flest GET - som godt kunne hatt path-param slik vi ser i frontend-adresselinje, et antall POST - der noen få kunne vært put/delete, kun 1 PUT (anti-pattern for opprettBehandling er langt fra idempotent) 
Diskusjonen mellom pragmatikere/Get+Post og purister/GPPP(atch)D har gått i over 20 år :-)